packages/sources/elwood/src/transport/crypto.ts

```diff
@@ -90,16 +97,37 @@ export const transport: WebSocketTransport<WsTransportTypes> =
     override async sendMessages(
       context: EndpointContext<WsTransportTypes>,
       subscribes: SubscribeRequest[],
-      unsubscribes: SubscribeRequest[],
     ): Promise<void> {
-      const messages = subscribes.concat(unsubscribes)
+      // All connections that share an API key have the same set of subscribed symbols so if there
+      // are multiple EAs that share the same key then we should only subscribe to new symbols. For
+      // the same reason, it is not safe to unsubscribe otherwise another EA expecting the symbol
```

Comment:
An EA will unsubscribe from an asset if it the subscription is stale, i.e. if the EA hasn't received a request for that subscription in X time: https://github.com/smartcontractkit/ea-framework-js/blob/main/src/transports/websocket.ts#L402-L404

How does this interact with the above comment - Will this result in EA's unsubscribing from assets that other EA's are still using? Is it common to share a single set of API credentials across Elwood EAs?

Reply:
> Will this result in EA's unsubscribing from assets that other EA's are still using? Is it common to share a single set of API credentials across Elwood EAs?

Yes currently that is the behaviour which is obviously not great. Hence this PR makes it so that we never unsubscribe. This does mean that the EA will do extra work receiving data for assets that are not needed, but it seems better than the alternative.

Reply:
Sharing credentials is probably just a Chainlink only problem.